### PR TITLE
@alloy => Infinite analytics in new typed schema

### PR DESCRIPTION
**Pull request by eessex**
Starts mapping new typed analytics schema to analytics events related to infinite scroll a/b test.

Updates editorial constants `getEditorialHref` and `getFullEditorialHref`, which were returning incorrectly for "feature" and "standard" layouts, and adds some tests for them. 

Also adds a new `storiesOfPublishing` function, which wraps stories in tracking so that react-tracking events will be logged in the console.

Updates have been made based on conversation with @anipetrov

### CODE REVIEW

**Comment by artsyit** [comment]
Deploy preview for *artsy-reaction* ready!

Built with commit 94b75298d8dcca584ef05d44a5d30d7527897295

https://deploy-preview-1040--artsy-reaction.netlify.com

File: src/__stories__/storiesOf.js [hunk 1]
```diff
@@ -11,3 +13,23 @@ export function storiesOf(desc, mod) {
...
+
+export function storiesOfPublishing(desc, mod) {
+  return _storiesOf(desc, mod).addDecorator(storyFn => {
+    return (
+      <PublishingBoot>
+        {storyFn()}
+      </PublishingBoot>
+    )
+  })
+}
```
Comment (orta) [inline src/__stories__/storiesOf.js]: nice idea 👍

File: src/Components/Publishing/RelatedArticles/RelatedArticlesPanelLink.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  }
+
+  @track(props => ({
+    action: Schema.ActionType.Click,
+    action_name: Schema.ActionName.ArticleImpression,
+    subject: "Related article",
+    destination_path: getArticleHref(props.article.slug),
+    // TODO: Check where type & flow fit into new schema
+    flow: "article",
+    type: "thumbnail",
```
[the remark this answers is not in the document]
Comment (eessex) [reply src/Components/Publishing/RelatedArticles/RelatedArticlesPanelLink.tsx]: Type is a new property that was introduced for this test, i believe it distinguished between a link in the body of an article text from a 'thumbnail' styled preview in the sidebar or footer.

File: src/Analytics/Schema/Values.ts [hunk 3]
```diff
@@ -50,12 +52,34 @@ export enum ActionType {
    * continuous hover mode.
    */
   Hover = "Hover",
+
+  /**
+   * A UI element was rendered in the viewport
+   *
+   * TODO: Check if this should be camelCase to conform to other namings
+   */
+  ArticleImpression = "article_impression",
```
[the remark this answers is not in the document]
Comment (eessex) [reply src/Analytics/Schema/Values.ts]: Yeah I think a generic impression event makes sense, and other properties will prove it happened inside an article.  cc @anipetrov

File: src/Analytics/Schema/Values.ts [hunk 4]
```diff
@@ -50,12 +52,34 @@ export enum ActionType {
...
+
+  /**
+   * A placeholder for an article pageview, not sent directly to segment
+   * this is intercepted in Force to trigger a page event
+   *
+   * TODO: Investigate sending page events from Reaction
+   */
+  ArticlePageview = "articlePageview",
```
[the remark this answers is not in the document]
Comment (eessex) [reply src/Analytics/Schema/Values.ts]: Agreed, doesn't make sense to scope to articles -- curious if there are any other instances where we are manually triggering pageviews from Reaction?  We do it twice for editorial, on expanding a standard article or expanding a news article from their respective infinite scroll feeds.

[the remark this answers is not in the document]
Comment (alloy) [reply src/Analytics/Schema/Values.ts]: Nope, not yet. We do have them in Emission, but for now in the Artist work decided to keep using Force for that for the time being.

What you probably want to look at is the `dispatchOnMount` option of react-tracking, which is what we use in Emission.

File: src/Components/Publishing/RelatedArticles/RelatedArticleCanvasLink.tsx [hunk 5]
```diff
@@ -14,28 +15,26 @@ export interface RelatedArticleFigureData {
...
-    tracking.trackEvent({
-      action: "Clicked article impression",
-      destination_path: href,
-      entity_id: article.id,
-      entity_type: "article",
-      flow: "article",
-      impression_type: "Further reading",
-      type: "thumbnail",
-    })
+  @track(props => ({
```
[the remark this answers is not in the document]
Comment (eessex) [reply src/Components/Publishing/RelatedArticles/RelatedArticleCanvasLink.tsx]: This gives me an error, `Expected 0 type arguments, but got 1`

File: src/Components/Publishing/Layouts/StandardLayout.tsx [hunk 6]
```diff
@@ -1,6 +1,9 @@
 import { color, space } from "@artsy/palette"
+import * as Schema from "Analytics/Schema"
+import { getEditorialHref } from "Components/Publishing/Constants"
 import { get, omit } from "lodash"
 import React from "react"
+import track from "react-tracking"
```
Comment (alloy) [inline src/Components/Publishing/Layouts/StandardLayout.tsx]: Did you on purpose import this one instead of `track` from `"Analytics"` ?

File: src/Components/Publishing/RelatedArticles/RelatedArticleCanvasLink.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (alloy) [reply src/Components/Publishing/RelatedArticles/RelatedArticleCanvasLink.tsx]: Ah yes, you got to import our typed version of the `track` function:

```ts
import { track } from "Analytics"
```

File: src/Components/Publishing/RelatedArticles/RelatedArticlesPanel.tsx [hunk 7]
```diff
@@ -26,8 +22,10 @@ export class RelatedArticlesPanel extends React.Component<
   }
 
   @track(() => ({
-    action: Schema.ActionName.ArticleImpression,
-    subject: "Related articles",
+    action: Schema.ActionType.Impression,
+    // TODO: reevalutate double naming
+    context_module: Schema.Subject.RelatedArticles,
```
Comment (alloy) [inline src/Components/Publishing/RelatedArticles/RelatedArticlesPanel.tsx]: I’d expect this annotation to be at the class level:

```tsx
@track({ context_module: Schema.Subject.RelatedArticles })
 export class RelatedArticlesPanel extends React.Component<
   RelatedArticlesPanelProps
 > {
  // ...
}
```

**Comment by alloy** [comment]
Looks great! A few tiny comments, but mostly I realised that you weren’t updating the `track` imports to use our schema typed version from `"Analytics"`, so that should still happen. (Hopefully it won’t lead to too much required changes 🤞)

File: src/Components/Publishing/Layouts/StandardLayout.tsx (hunk 6 above)
Comment (eessex) [reply src/Components/Publishing/Layouts/StandardLayout.tsx]: Oh cool- didn’t realize that was a thing, had remembered you said to use library rather than removed utils version, but sounds like that was only for untyped calls.  Will update!

Comment (alloy) [reply src/Components/Publishing/Layouts/StandardLayout.tsx]: Yeah sorry about that, I should have written that up for you!

File: src/Analytics/Schema/Values.ts [hunk 8]
```diff
@@ -50,6 +52,16 @@ export enum ActionType {
...
+
+  /**
+   * A UI element was rendered in the viewport
+   */
+  Impression = "Impression",
+
+  /**
+   * Triggers a pageview in force, skips segment
+   */
+  ClickedReadMore = "Clicked read more",
```
Comment (l2succes) [inline src/Analytics/Schema/Values.ts]: @eessex Why did you need to create a new `ActionType` here instead of using `Click`?

Comment (eessex) [reply src/Analytics/Schema/Values.ts]: This string maps to an [override in Force here](https://github.com/artsy/force/blob/dabc83fef5a266c14f6df9a4163023ba307c40d3/src/desktop/assets/analytics.coffee#L16) -- we will look at implementing segment's pageview in Reaction in the future, but for now this keeps the pageview firing.

Comment (l2succes) [reply src/Analytics/Schema/Values.ts]: I see, thanks for the context
